File: Lesson_5/Class_work/Task_1/CalculatorPresenter.java
package Lesson_5.Class_work.Task_1;

public class CalculatorPresenter {
    private CalculatorView view;
    private CalculatorModel model;

    public CalculatorPresenter(CalculatorView view, CalculatorModel model) {
        this.view = view;
        this.model = model;
    }

    public void performCalculation() {
        double number = view.getDouble();
        char operator = view.getInputOperator();
        view.scan.nextLine();//строка, без которой, почему-то второй getDouble сразу закрывает скан, попадает в кетч и только после этогьо нормально захватывает
        double number_2 = view.getDouble_2();
        model.setResult(number);

        switch (operator) {
            case '+':
                model.add(number_2);
                break;
            case '-':
                model.minus(number_2);
                break;
            case '*':
                model.mult(number_2);
                break;
            case '/':
                model.division(number_2);
                break;


        }

        view.displayResult(model.getResult());

    }
}

File: Lesson_7/Task_3/CalculatorPresenter.py
from CalculatorModel import  CalculatorModel
from  CalculatorView import CalculatorView

class CalculatorPresenter :

    def __init__(self, model_1, view_1):
        self.model = model_1
        self.view = view_1



    def performCalculation(self):
        number =CalculatorView.getDouble()
        operator = CalculatorView.getInputOperator()
        number_2 = CalculatorView.getDouble()
        CalculatorModel.setResult(number)

        if operator=="+":
            CalculatorModel.add(number_2)

        if operator=="-":
            CalculatorModel.minus(number_2)

        if operator=="*":
            CalculatorModel.mult(number_2)

        if operator=="/":
            CalculatorModel.division(number_2)
        CalculatorView.displayResult(CalculatorModel.getResult())



